Next 16

Fridus opened the pull request.

Fridus committed:
Try other headers

Fridus committed:
Nextjs 16

Fridus committed:
debug

Fridus committed:
Double decode param url

Fridus commented:
@LoicMahieu Après la version next 15.1.6, je constate un comportement différent sur vercel pour le parsing des chemins dynamique `/viewer/[url]` . Le paramètre est partiellement décodé. Ce qui dans notre cas pose problème car ici, l'url était décodée puis utilisée pour fetch un fichier chez google. La signature était donc impactée. Ré encoder les paramètres ne fonctionne pas non plus car ça ré-encodait trop d'éléments...

Je te laisse juger de la solution

LoicMahieu commented:
## Version EML

#### Direct sur googleapis:

- https://eml-viewer-igloo.vercel.app/viewer/https%3A%2F%2Fstorage.googleapis.com%2Fstable-next-m6bt8p-storage%2F7ca898fa43aa418eae5ead214384e926%2Fea40da50368245e7b641ecd39342ad8d.eml%3FGoogleAccessId%3Dsa-le-next-m6bt8p-storage-gcs%2540foldio-staging.iam.gserviceaccount.com%26Expires%3D2081974134%26Signature%3DjzBgtRINHvkYaW%252F%252FgsuNQxN%252BoeE1Uudsp%252BJXwgQps%252BT5Ia%252BXaKSEsxFShav%252B11W%252B5bcMMKvDvGWWTtPLd%252FWVDAwn%252FF2qXaAYrV9uPEASAX%252B9zdgIATXRSjV01TySuMRqFBJtWWqJ%252FQ9jDx1V%252FNiH6csdB2BoA3PoTMSj9wsWU5U3cequidrFAPVwTemZ99agkGEBPprtajk9KRKvo15oPl39c1UkA0gzGbGQGv2%252FcJqmvL%252F3izwzLLRaDjA1E1dL9nVZmOUH6%252FB7y3X9HqYZfGrqwpvJQl9Trdt8Kw%252BT0HgVGpmh%252FxM9NVScuxn1m9jOUzhtT0q0heNKo36rVsULTw%253D%253D%26response-content-disposition%3Dinline%253B%2520filename%253D%2522Proposition%2520de%2520rendez-vous.eml%2522


#### Direct sur Foldio:

- https://eml-viewer-igloo.vercel.app/viewer/https%3A%2F%2Fnext.foldio.app%2Fapi-v2%2Fstorage-files%2Fdownload%3Fid%3D5987%26signature%3DeyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NTk4NywiaWF0IjoxNzU5Mzk5NzU1LCJleHAiOjIwNzQ5NzU3NTV9.6yzxX1h3yxs6P6aGpVotBdLDsxXuOAXSj7uEgUqvy3w%26contentDispositionType%3Dinline


## Version MSG

#### Direct sur googleapis:

- https://eml-viewer-igloo.vercel.app/viewer/https%3A%2F%2Fstorage.googleapis.com%2Fstable-next-m6bt8p-storage%2F51783d455e484710bba8a65099859cb0%2F0ad5273780ed49dc9d5c1c3b1ee2ab6b.msg%3FGoogleAccessId%3Dsa-le-next-m6bt8p-storage-gcs%2540foldio-staging.iam.gserviceaccount.com%26Expires%3D2083800976%26Signature%3DK2FZ4NFNyNq%252FJAUObiAqGHinW2OC6wggB%252FqJfbqfqBH6cx0LqXl4TpkvFLkZst8IPWxknO9Te2K9nVAUl6e3PdKsHIIW7bmjmfHkem93oilKmfrPTUF1OBkYxzA%252F5IQr6Nee8yuwtoFFRG8JfF%252FRPVqvh%252FlvxsJoS6%252FDAw0pmqfE7MLqY3z0vt9c9Ke8RhkDFO58274H%252Fd9MW319%252Fc%252Bss%252Fu0ZO5IO7df7YhsqmK74uOjpNMjwD3n2N%252BobnYxWQrEqA%252FtcT%252B9O7Di%252FHE69MS%252BQrH2sc%252BIf4Hizw%252BMexxwz2YM9KZaI5ezMoRUIk6o7QENg3bkxVswIwA0evMD9nRxgg%253D%253D%26response-content-disposition%3Dinline%253B%2520filename%253D%2522Sample%2520MSG%2520File.msg%2522/msg

LoicMahieu committed:
Implement email viewer components and update URL handling

- Updated the URL handling in the form submission to use query parameters.
- Added new components: EmailParser, EmailViewer, ErrorView, and layout for the email viewer.
- Introduced mailparser utility for parsing email content.
- Created viewer pages for handling different email types (eml and msg) with appropriate loading states.
- Ensured proper structure for displaying email content, including attachments and headers.
